src/actions/FavoritesActions.js

```diff
@@ -0,0 +1,64 @@
+import firebase from '@firebase/app';
+import '@firebase/firestore'
+
+import {
+    FAVORITE_ADDED,
+    FAVORITE_ADD_FAILED,
+    FAVORITE_REMOVED,
+    FAVORITE_REMOVE_FAILED
+} from './types';
+
+export const attemptToAddFavorite = ({selectedSite, currentUser}) => {
+
+    return async (dispatch) => {
+        const favoriteMetadata = {
+            title: selectedSite.title,
+            id: selectedSite.id,
+        };
+
+        return firebase.firestore().doc(`users/${currentUser.email}/favorites/${selectedSite.title}`)
+            .set(favoriteMetadata)
+            .then(() => {
+                dispatch({
+                    type: FAVORITE_ADDED,
+                    payload: {favoriteToAdd: selectedSite}
+                });
+            })
+            .catch(error => {
+                dispatch({
+                    type: FAVORITE_ADD_FAILED,
+                    payload: {error}
+                });
+
+            });
+    };
+};
```

Comment:
The add and remove actions are nearly identical. Try to add the favorite to the user's `favorites` collection in firebase. The great thing about firebase is that we don't need two cases for if 1) the favorites collection exists and 2) the favorites collection doesn't exist. We just say "add this to the collection of favorites" and either it adds it to the existing collection, or creates a new collection and makes this the first entry. 

For the local app: Since we have all the site's info (because we were passed `selectedSite` from `SiteDetailScreen`) we can pass the full object down to the reducers that will update `currentUser`. 

For firebase: To keep the database from getting too large with duplicate data, the firebase favorites collection only keeps the metadata about a favorite (`title` and `id`). Here when we add the favorite, we pass the full info to the rest of the app... but when a user logs in and we need to grab their favorites, we use this firebase metadata and pair it with the incoming full site info to create the full favorite objects for the user. More on how we do that in `AuthReducer`.